Merge del branch DEV nel MAIN

JohnnyLAmpAz opened the pull request:
Con questa PR proporrei di approvare quanto abbiamo fatto fino ad ora mergiando nel main e di eliminare il branch _GeneralDevelopment_. Così, **invece che unire i nostri personali in _GeneralDevelopment_ e solo successivamente nel main, gestiamo il tutto con delle _pull request_ direttamente sul main.**

Cosa ne dite?

DavideBrambilla98 committed:
aggiunto nel pom il miglayout

DavideBrambilla98 committed:
Aggiunta classe OrderModel

Implementata la classe OrderModel nella quale vengono inseriti tutti i parametri dell'ordine (tranne i prodotti e le relative quantità) nel modello.

DavideBrambilla98 committed:
Aggiunta classe ProductOrderModel

Classe che uso per associare i prodottti agli ordini
sono ancora da implementare alcuni metodi!

DavideBrambilla98 committed:
Mod non presa nel commit precedente

DavideBrambilla98 committed:
Merge remote-tracking branch 'origin/GeneralDevelopment' into OrderModelBranch

DavideBrambilla98 committed:
Modifiche al modello per la gestione degli ordini

- Aggiunti metodi toString agli enumeratori relativi agli ordini
- Cancellata la classe ProductOrderModel poichè gestisco il record della tabella
   ProdottiOrdini direttamente nel modello "OrderModel"
- Aggiunti alcuni dei metodi per la gestione dei record "ProdottiOrdini"

MassimilianoGervasoni committed:
creazione classe BoxModel

JohnnyLAmpAz committed:
DB improvements

- renamed password field in UTENTE table
- untracked db file
- Db class to accept different db file path (for future test purposes)

JohnnyLAmpAz committed:
Box: fix isValid() + renamed field using ASCII

JohnnyLAmpAz committed:
fix BaseModel after Db improvements

DavideBrambilla98 committed:
OrderModel Class modified

- Aggiunti metodi che permettono di gestire la "listaProdottiOrdiniRecord" e i record
  che contiene.
- Modificata la visibilità da private a package di alcuni metodi della classe
  ProductModel, poichè mi sono serviti nella classe OrderModel

JohnnyLAmpAz committed:
First version of UtenteModel

MassimilianoGervasoni committed:
Update BoxModel

aggiunta descrizioni e correzione degli update dei record

JohnnyLAmpAz committed:
bozza note: disegno box

JohnnyLAmpAz committed:
typo in UtenteModel

JohnnyLAmpAz committed:
Fix and update UtenteModel

DavideBrambilla98 committed:
Aggiornamento classe OrderModel e StatoOrdine

- Aggiunto metodo che permette di modificare lo stato dell'ordine
- Aggiunta JavaDoc dei vari metodi contenuti nella classe OrderModel
- Corretto metodo getListaProdottiFromDb
- Modificato enumeratore StatoOrdine

JohnnyLAmpAz committed:
implemented Login:

UI Dialog for login
fixed and updated Db and UtenteModel

DavideBrambilla98 committed:
Update OrderModel.java

DavideBrambilla98 committed:
Merge pull request #8 from JohnnyLAmpAz/login-lorenzo

Utente Model + Login + minor updates

JohnnyLAmpAz committed:
moved LoginDialog in new package smartmag.ui

JohnnyLAmpAz committed:
Merge branch 'login-lorenzo' of github.com:JohnnyLAmpAz/smartmag into login-lorenzo

MassimilianoGervasoni committed:
aggiunta documentazione BoxModel

JohnnyLAmpAz committed:
Revert check if passwd of Utente is blank

DavideBrambilla98 committed:
Modifiche classe OrderModel

Effettuate le modifiche suggerite nei commenti della PR#9

JohnnyLAmpAz committed:
Comparable Utente + TreeMap in the Model

JohnnyLAmpAz committed:
fix UtenteModel

loadAllUsers() -> fetchAllUsers() that only returns the TreeMap and
doesn't assign it to instances

DavideBrambilla98 committed:
Update OrderModel.java

Aggiornato metodo ordineFromOrdineRecord:
adesso il metodo ritorna l'ordine completo (con lista prodotti e qta)

JohnnyLAmpAz committed:
ChangeListener support for BaseModel

this way, any Model we create will be able to invoke
notifyChangeListeners in order to notify any sort of state change
happened

DavideBrambilla98 commented:
Per me va bene fare il merge.
La classe OrderModel è già stata aggiornata sulla base della review alla PR #9 
La classe BoxModel mi sembra ok. Mancano le modifiche richieste nella review alla PR #10  ma si possono effettuare anche dopo il merge volendo dato che non sono invasive.

MassimilianoGervasoni commented:
sto apportando ora le modifiche alla classe BoxModel, ho quasi finito  devo solo creare il metodo che restituisce le istanze dal database con la treeMap; per il resto sono d'accordo

MassimilianoGervasoni committed:
correzioni BoxModel

correzioni in base a review

JohnnyLAmpAz committed:
Utente clone() method + getUtente() on the model

JohnnyLAmpAz committed:
UtenteModel: put new inistances in the hashmap only in constructor

JohnnyLAmpAz committed:
Update UtenteModel:

fix constructor, init static map on load, new getAllUserModels() to get
all UserModel instances

DavideBrambilla98 committed:
Merge pull request #9 from JohnnyLAmpAz/OrderModelBranch

Order model

DavideBrambilla98 committed:
Modificata classe OrderModel

Aggiunto metodo clone
Aggiunta TreeMap

DavideBrambilla98 committed:
Merge branch 'OrderModelBranch' into GeneralDevelopment

DavideBrambilla98 committed:
Errore dopo il merge nell' enum

JohnnyLAmpAz committed:
Merge branch 'login-lorenzo' into GeneralDevelopment

JohnnyLAmpAz committed:
Fix OrderModel constructor

JohnnyLAmpAz committed:
Suppress casting warning

JohnnyLAmpAz committed:
Created UsersTableModel file, not implemented yet

MassimilianoGervasoni committed:
Update BoxModel

aggiornamento classe BoxModel con aggiunta metodo getAllBoxModels

JohnnyLAmpAz committed:
Merge pull request #10 from JohnnyLAmpAz/BoxModel-Massimiliano

Box model
